### PR TITLE
fix: removed `try_join` to await futures

### PR DESCRIPTION
**Pull request by frolvanya**
This PR fixes a misconception behind `try_join` method. It doesn't await until the finish state of all futures, but rather kills other futures if one one them returns an error. You can try it yourself:

```rust
use anyhow::Result;

async fn a() -> Result<()> {
    println!("Throwing an error");
    anyhow::bail!("This is an error");
}

async fn b() -> Result<()> {
    tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
    println!("Returning OK");
    Ok(())
}

#[tokio::main]
async fn main() -> Result<()> {
    futures::try_join!(a(), b())?;

    Ok(())
}
```
As you can see it doesn't print "Returning OK" to the stdout:
image

Now let's modify our code to use regular `join_all` instead and handle possible errors afterwards:
```rust
use anyhow::Result;
use futures::FutureExt;

async fn a() -> Result<()> {
    println!("Throwing an error");
    anyhow::bail!("This is an error");
}

async fn b() -> Result<()> {
    tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
    println!("Returning OK");
    Ok(())
}

#[tokio::main]
async fn main() -> Result<()> {
    futures::future::join_all([a().boxed(), b().boxed()])
        .await
        .into_iter()
        .collect::<Result<_>>()?;

    Ok(())
}
```

And here's the new ouput which waits for both function to finish their execution before handling an error:
image

### CODE REVIEW

File: rpc-server/src/modules/blocks/mod.rs
Comment (frolvanya) [inline rpc-server/src/modules/blocks/mod.rs:298]: I'd like to simplify this more, but I'm out of ideas. They have different return types, so I can't use regular `join_all`. Please, let me know if you have any ideas

File: rpc-server/src/modules/queries/methods.rs
Comment (frolvanya) [inline rpc-server/src/modules/queries/methods.rs:269]: This one is also a bit messy

**Comment by frolvanya** [comment]
@race-of-sloths

**Comment by race-of-sloths** [comment]
@frolvanya Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Great job! Usain Bolt jealous of your results!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/frolvanya?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/frolvanya?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/frolvanya)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/frolvanya)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: rpc-server/src/utils.rs
Comment (frolvanya) [inline rpc-server/src/utils.rs:599]: This fixes bad naming practice, but it can cause some issues with metrics if it relies on old name

File: rpc-server/src/modules/blocks/mod.rs
Comment (kobayurii) [reply rpc-server/src/modules/blocks/mod.rs:298]: This is done once when the service starts. Any drop at this point should not prevent the service from starting. So I don't think we need to guarantee that all tasks are completed here.

File: rpc-server/src/modules/queries/methods.rs
Comment (kobayurii) [inline rpc-server/src/modules/queries/methods.rs:253]: We don't have to wait for both tasks to be completed. The failure of any task should return a error immediately.

File: rpc-server/src/modules/queries/methods.rs
Comment (kobayurii) [inline rpc-server/src/modules/queries/methods.rs:259]: same as above.

**Review by kobayurii — CHANGES_REQUESTED** [review]
Overall it looks good. Left a couple of comments. Thanks

**Comment by frolvanya** [comment]
> Overall it looks good. Left a couple of comments. Thanks

Fixed. Thanks for the review!

**Review by kobayurii — APPROVED** [review]
Thank you!

**Comment by race-of-sloths** [comment]
🥁  __Score it!__

@kobayurii, please score the PR with `@race-of-sloths score [1/2/3/5/8/13]`. The contributor deserves it.
If no scoring is provided within 24 hours, this PR will be scored as 2 🦥

**Pull request merged**
